crates/chain/src/tx_graph.rs

```diff
@@ -590,11 +591,20 @@ impl<A: Anchor> TxGraph<A> {
         let tx_node = self.txs.entry(txid).or_default();
         match tx_node {
             TxNodeInternal::Whole(existing_tx) => {
-                debug_assert_eq!(
-                    existing_tx.as_ref(),
-                    tx.as_ref(),
-                    "tx of same txid should never change"
-                );
+                // We want to be able to replace an unsigned tx with a signed tx.
+                // The tx with more weight has precedence (and tiebreak with the actual tx data).
+                // We can also check whether the witness is valid and also prioritize signatures
+                // with less weight, but that is more work and this solution is good enough.
```

Comment:
Prediction: It's less lines of code than this comment to just do it correctly.

Reply:
Changed it so we zip across both tx inputs and compare witnesses. We prioritize non-empty witnesses. If witnesses both have data, we prioritize the witness that is smaller. If both witnesses are the same size, we prioritize based on lexicographical order.